Include/internal/pycore_lock.h

```diff
@@ -213,6 +213,41 @@ _PyOnceFlag_CallOnce(_PyOnceFlag *flag, _Py_once_fn_t *fn, void *arg)
     return _PyOnceFlag_CallOnceSlow(flag, fn, arg);
 }
 
+// A readers-writer (RW) lock. The lock supports multiple concurrent readers or
+// a single writer. The lock is write-preferring: if a writer is waiting, then
+// new readers will be blocked. This avoids starvation of writers.
```

Comment:
It might be nice to also briefly note that another name for this is a "shared mutex", with distinct "shared" vs. "exclusive" operations.  (Basically, what you said in https://github.com/python/cpython/pull/112471#discussion_r1418183599.)

Reply:
I added a comment.